xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/table/livetable.js

```diff
@@ -452,6 +452,13 @@ XWiki.widgets.LiveTable = Class.create({
           }
           container.update(text.escapeHTML());
         }
+
+        // When a cell is part of a non-viewable row, we display a content indicating that the content is not present
+        // because the current user does not have the right to view it.
+        if (!row['doc_viewable'] && !row[column]) {
+          td.update($(table.domNodeName + '-inaccessible-cell').innerHTML + '<sup>*</sup>')
+        }
```

Comment:
@mflorea Does this change looks alright to you? Also, I'm not sure of the best way to test this change, wdyt is the right way to do it?

Reply:
You need to use ``fieldName`` instead of ``column``. Then the field value can be boolean (or empty string), so checking for ``!row[fieldName]`` is not right. You need to either check if the row has the key, using ``hasOwnProperty``, or check for ``null`` or ``undefined`` explicitly.

Moreover, if you replace the content of the cell entirely then the execution of the previous code is useless. It doesn't make sense to try to compute the link or the cell text (including special handling of the ``doc.name`` column when the row is not viewable) if in the end you overwrite all that. This code needs a bit more refactoring.